utils/utils.go

```diff
@@ -0,0 +1,397 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Each resource request will be fetched from the Kubernetes API Server at a
+// max of 20 retries, with 15 seconds in between each attempt.
+// This creates a max wait time of upto 5 minutes that a resource request
+// must successfully return within, before moving on.
+
+// MaxRetries is the maximum number of retries that a resource will be
+// attempted to be fetched from the Kubernetes API Server.
+const MaxRetries = 20
+
+// RetryInterval is the number of seconds to sleep in between requests
+// to the Kubernetes API Server.
+const RetryInterval = 15
+
+// RunEKSSmokeTest instantiates the EKS Smoke Test.
+func RunEKSSmokeTest(t *testing.T, resources []apitype.ResourceV3, kubeconfigs ...interface{}) {
+	// Map the cluster name to the total desired Node count across all
+	// NodeGroups.
+	clusterNodeCount, err := MapClusterToNodeCount(resources)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Map the cluster name to the KubeAccess client-go tool bag.
+	kubeAccess, err := MapClusterToKubeAccess(kubeconfigs...)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Run the smoke test against each cluster, expecting the total desired
+	// Node count.
+	for clusterName := range kubeAccess {
+		printAndLog(fmt.Sprintf("Testing Cluster: %s\n", clusterName), t)
+		clientset := kubeAccess[clusterName].Clientset
+		EKSSmokeTest(t, clientset, clusterNodeCount[clusterName])
+	}
+}
+
+// EKSSmokeTest runs a checklist of operational successes required to deem the
+// EKS cluster as successfully running and ready for use.
+func EKSSmokeTest(t *testing.T, clientset *kubernetes.Clientset, desiredNodeCount int) {
+	// Print API Server info.
+	APIServerVersionInfo(t, clientset)
+
+	// Run all tests.
```

Comment:
I’ll admit I had assumed we would do something a little more course grained - like just deploying a workload into the cluster and validating that it works (including can allocate LB, can allocate storage, can respond correctly to network traffic, etc). 

That may be an additional thing on top of what’s here.

Reply:
You're assumption is valid, and doing workload testing post cluster readiness is definitely something I too wanted to add - this PR just took longer than expected to test the readiness piece, so I did not get around to it. I've gone ahead and added it to the test coverage todo being tracked in https://github.com/pulumi/pulumi-eks/issues/124.